src/View/Cell.php

```diff
@@ -160,14 +179,31 @@ public function render($template = null)
         if ($template === null) {
             $template = $this->template;
         }
-        $builder = $this->viewBuilder();
-        $builder->layout(false);
-        $builder->template($template);
 
         $cache = [];
         if ($this->_cache) {
-            $cache = $this->_cacheConfig($template);
+            $cache = $this->_cacheConfig($this->action);
+            $result = Cache::read($cache['key'], $cache['config']);
+            if ($result !== false) {
+                return $result;
+            }
+        }
+
+        try {
+            $reflect = new ReflectionMethod($this, $this->action);
+            $reflect->invokeArgs($this, $this->args);
+        } catch (ReflectionException $e) {
+            throw new BadMethodCallException(sprintf(
+                'Class %s does not have a "%s" method.',
+                get_class($this),
+                $this->action
+            ));
         }
+
+        $builder = $this->viewBuilder();
+        $builder->layout(false);
+        $builder->template($template);
```

Comment:
Could all of this go in the `$render` closure? `View->cache()` also checks the cache key.